src/stores/useHolochainStore.js

```diff
@@ -104,21 +106,107 @@ const makeUseHolochainStore = ({ installed_app_id, app_ws_url, is_hpos_served, h
 
       return result
     },
+    async zomeCall(args) {
+      const zomeCallArgs = {
+        appId: installed_app_id,
+        roleId: args.role_name,
+        zomeName: args.zome_name,
+        fnName: args.fn_name,
+        payload: args.payload
+      }
+
+      const response = await this.hposHolochainCall({path: 'zome_call', headers: {}, params: zomeCallArgs})
+      return response
+    },    
     setCredentials(cellId) {
       this.signingCredentials = new Promise(async (resolve, reject) => {
-        try {
-          const adminWs = await AdminWebsocket.connect(`ws:localhost:${hc_admin_port}`)
-          await adminWs.authorizeSigningCredentials(cellId)
-        } catch(e) {
-          console.log(`holochainCallZome error authorizeSigningCredentials AdminWebsocket: ws:localhost:${hc_admin_port}`, e)
-          reject()
+        if( !is_hpos_served ) { // If running a raw holochain we need to authorize zome calls once
+          try {
+            const adminWs = await AdminWebsocket.connect(`ws:localhost:${hc_admin_port}`)
+            await adminWs.authorizeSigningCredentials(cellId)
+          } catch(e) {
+            console.log(`holochainCallZome error authorizeSigningCredentials AdminWebsocket: ws:localhost:${hc_admin_port}`, e)
+            reject()
+          }
+
+          resolve()
+        } else {
+          try {
+            const [keyPair, signingKey] = await generateSigningKeyPair()
+            const params = { cellId, signingKey }    
+            const cap_token = await this.hposHolochainCall({path: 'cap_token', headers: {}, params})
+
+            const signingCredentials = {
+              capSecret: new Uint8Array(listify(cap_token, (_, value) => (Number(value)))),
+              keyPair,
+              signingKey
+            }
+
+            await setSigningCredentials(cellId, signingCredentials)
+          } catch (e) {
+            console.log(`Error setting signing credentials`, e)
+            reject()
+          }
+
+          resolve()
```

Comment:
also, the result of this promise is being assigned to `this.signingCredentials` (line 122), but the promise never returns anything and instead calls various state mutating functions (either `adminWs.authorizeSigningCredentials` or `setSigningCredentials`.
Given that, what's the purpose of `this.signingCredentials` and why does this need to be a promise and not just inline `await`s?

Reply:
You're correct, I removed it.